### PR TITLE
MULE-13188: ValueProviderService creates a connection although this m…

### PR DESCRIPTION
**Pull request by estebanwasinger**
…ay not be required

### CODE REVIEW

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/enricher/ValueProvidersParameterDeclarationEnricher.java [hunk 1]
```diff
@@ -132,26 +135,32 @@ private void enrichContainerModel(ParameterizedDeclaration<?> containerDeclarati
...
+        ValueProviderFactoryModelProperty.builder(resolverClass.value());
+    ParameterizableTypeWrapper resolverClassWrapper = new ParameterizableTypeWrapper(resolverClass.value());
     List<ExtensionParameter> resolverParameters = resolverClassWrapper.getParametersAnnotatedWith(Parameter.class);
 
     resolverParameters.forEach(param -> propertyBuilder
         .withInjectableParameter(param.getName(), param.getMetadataType(typeLoader), param.isRequired()));
 
-    enrichWithConnection(propertyBuilder, resolverClassWrapper);
-    enrichWithConfiguration(propertyBuilder, resolverClassWrapper);
+    AtomicBoolean requiresConfiguration = new AtomicBoolean(false);
```
Comment (marianogonzalez) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/enricher/ValueProvidersParameterDeclarationEnricher.java]: this code is synchronous. Use Reference instead

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/config/ConfigurationProviderToolingAdapter.java [hunk 2]
```diff
@@ -185,6 +190,27 @@ private MetadataContext getMetadataContext() throws MetadataResolvingException,
...
+    ConnectionProvider<?> connectionProvider = configuration.getConnectionProvider()
+        .orElseThrow(() -> new ValueResolvingException("Unable to obtain the Connection Provider Instance", UNKNOWN));
+
+    ConnectionProvider unwrap = unwrap(connectionProvider);
+    ConnectionProviderModel connectionProviderModel =
+        getConnectionProviderModel(unwrap.getClass(), getAllConnectionProviders(getExtensionModel(), getConfigurationModel()))
+            .orElseThrow(() -> new ValueResolvingException("Internal error. Unable to obtain the Connection Provider Model",
+                                                           UNKNOWN));
+
+    return info.execute(unwrap, connectionProviderModel);
```
Comment (marianogonzalez) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/config/ConfigurationProviderToolingAdapter.java]: something called info shouldn't have an execute method

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/value/ValueProviderUtils.java [hunk 3]
```diff
@@ -59,4 +63,17 @@ static ValueBuilder cloneAndEnrichValue(Value value, Map<Integer, String> partOr
...
 
+  /**
+   * Given a list of {@link ParameterModel} retrieves all the parameters that have an associated {@link ValueProviderModel}
+   *
+   * @param parameterModels Parameters to introspect
+   * @return The {@link List} of {@link ParameterModel} that have an associated {@link ValueProviderModel}
+   */
+  public static List<ValueProviderModel> getValueProviderModels(List<ParameterModel> parameterModels) {
+    return parameterModels.stream()
+        .map(ParameterModel::getValueProviderModel)
```
Comment (marianogonzalez) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/value/ValueProviderUtils.java]: flatMap

File: modules/spring-config/src/main/java/org/mule/runtime/config/spring/internal/LazyComponentInitializer.java [hunk 4]
```diff
@@ -26,4 +29,15 @@
...
+  /**
+   * Calling this method guarantees that the requested component from the configuration will be created.
+   * <p/>
+   * The requested component must exists in the configuration.
+   *
+   * @param location the location of the configuration component.
+   * @throws MuleRuntimeException if there's a problem creating the component or the component does not exists.
+   * @return The {@link List} of created component's names
+   */
+  List<String> createComponent(Location location);
```
Comment (marianogonzalez) [inline modules/spring-config/src/main/java/org/mule/runtime/config/spring/internal/LazyComponentInitializer.java]: weird. If the method is called createComponent, I expect it to return the component. Plus, why a List? how can this create many components?

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/value/ValueProviderUtils.java (hunk 3 above)
Comment (estebanwasinger) [reply modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/value/ValueProviderUtils.java]: Why a flat map? I don't get it

**Pull request merged**